test/src/helpers.cc

```diff
@@ -68,22 +73,111 @@ void check_partitions(
   for (const auto& p : partitions) {
     CHECK(!partitioner.done());
     CHECK(!unsplittable);
+    //TBD: *if* previous loop iteration next() happened to finish, following (next loop iteration) ...ok() will still return true,
+    //(as Status::Ok() is returned from top when already 'done()' on entry to next())
+    //tho suppose that partition check following (for actual data partition data) should fail, assuming the two last partition entries
```

Comment:
If a check fails, we don't care about what happens after that. It's a failed test either way.